pq-src/build.rs

```diff
@@ -309,44 +336,48 @@ fn main() {
             LIBPQ_BASE.to_vec(),
         )
     };
-    // don't even try on windows it's not there
-    if std::env::var("CARGO_CFG_WINDOWS").is_err() {
-        conditional_define(TEST_FOR_STRCHRNUL, "HAVE_STRCHRNUL", &mut basic_build);
-    }
+
+    // Check if strchrnul and/or strsignal are supported, if so add defines for them
+    conditional_define(TEST_FOR_STRCHRNUL, "HAVE_STRCHRNUL", &mut basic_build);
```

Comment:
Why does this change fail windows builds? I would expect that the test-compilation that conditional define does for strchrnul fails on windows and HAVE_STRCHRNUL not to be set. But I think it is getting set. I have to look into this again :thinking:

Reply:
The problem is that the windows C(++) compiler does not treat this implict declaration as an error. 
It only prints
`warning C4013: 'strchrnul' undefined; assuming extern returning int` 
and exist normally

One can enable this specific warning to cause an error with the `we4013` flag. I think it would make sense to add/enable that flag for windows builds. What do you think @weiznich?

Reply:
Done in `Refine check_compiles function`